cg/src/domains/subjects/subjects.service.js

```diff
@@ -28,28 +32,51 @@ class SubjectsService {
     });
   }
 
-  async registerConsentToProcessData(subjectId, personalData, processorIds = []) {
+  async registerConsentToProcessData(subjectId, personalData, processorsConsented = []) {
+    const [ subjectExists ] = await this.db('subjects')
+      .where('id', subjectId);
+    
+    if(subjectExists) throw new Unauthorized('Subject already gave consent, please use the update-consent endpoint');   
+    let processorIdsWithAddresses;
     await this.db.transaction(async trx => {
       await this._initializeUserInTransaction(trx, subjectId, personalData);
-
-      const processorIdsWithAddresses = await this._getProcessorIdsWithAddresses(trx, processorIds);
-
-      if (processorIdsWithAddresses.length !== processorIds.length) {
-        throw new ValidationError('Specified processor does not exist.');
+      processorIdsWithAddresses = await this._getProcessorIdsWithAddresses(trx, processorsConsented);
+      if (processorIdsWithAddresses.length !== processorsConsented.length) {
+        throw new ValidationError('At least one of the processors specified is not valid');
       }
-
-      if (processorIdsWithAddresses.some(p => !p.address)) {
+      if(processorIdsWithAddresses.some(p => !p.address)) {
         throw new ValidationError(
           `At least one of the processors doesn't have an address assigned`
         );
       }
-
       await Promise.all(
-        processorIds.map(processorId => this._setConsentGiven(trx, subjectId, processorId))
+        processorsConsented.map(processor => this._setConsentGiven(trx, subjectId, processor))
       );
+    });
+    await recordConsentGivenTo(subjectId, processorIdsWithAddresses.map(p => p.address));
+  }
 
-      await recordConsentGivenTo(subjectId, processorIdsWithAddresses.map(p => p.address));
+  async updateConsent(subjectId, processorsConsented = []){
+    const [ subjectExists ] = await this.db('subjects')
+      .where('id', subjectId);
+  
+    if(!subjectExists) throw new NotFound('Subject not found, please use the give-consent endpoint');
+    let processorIdsWithAddresses;
+    await this.db.transaction(async trx => {
+      processorIdsWithAddresses = await this._getProcessorIdsWithAddresses(trx, processorsConsented);
```

Comment:
can we share this validation logic with the above function? abstract it into a shared function call?